RepairShop-Frontend/src/components/ViewAppointments.vue

```diff
@@ -0,0 +1,149 @@
+<template>
+  <div id="ViewServices">
+    <h2>Your Appointments</h2>
+    <template>
+      <div>
+        <div v-if="errorViewServices">
+          <span v-if="errorViewServices" style="color: red">
+            {{ errorViewServices }}
+          </span>
+        </div>
+        <div v-else>
+          <b-table
+            :items="items"
+            :fields="fields"
+            :outlined="true"
+            :key="this.items.length"
+          >
+            <template #cell(cancellation)="row">
+              <b-button
+                size="sm"
+                v-on:click="cancelAppointment(row.item.ID)"
+                class="mr-2"
+                variant="danger"
+                v-if="checkIfWeekAhead(row.item.ID)"
+              >
+                Cancel Appointment
+              </b-button>
+              <b-button
+                size="sm"
+                v-on:click="cancelAppointment(row.item.ID)"
+                class="mr-2"
+                disabled
+                v-else
+              >
+                Cancel Appointment
+              </b-button>
+            </template></b-table
+          >
+        </div>
+      </div>
+    </template>
+  </div>
+</template>
+
+<script>
+import axios from "axios";
+import { CANCEL_APPOINTMENT_ENDPOINT } from "../constants/constants";
+var config = require("../../config");
+var AXIOS = axios.create({
+  baseURL: "http://" + config.dev.backendHost + ":" + config.dev.backendPort
+});
+
+export default {
+  data() {
+    return {
+      errorViewServices: "",
+      appointments: [],
+      fields: ["ID", "Service", "start", "end", "cancellation"],
+      items: [],
+      idToDateTimeMap: {}
+    };
+  },
+
+  //fetch all of this customer's appointments and display them in a table
+  created: function() {
+    this.getAppointments();
+  },
+  methods: {
+    // Should output something like "Tue Mar 02 2021 10:00:00 GMT-0500 (Eastern Standard Time)" given a timestamp
+    displayDateTime(dateTime) {
+      let date = new Date(dateTime).toString();
+      if (date == "Invalid Date") return "";
+      else
+        return (
+          date.slice(0, 10) +
+          ", " +
+          date.slice(11, 15) +
+          " at " +
+          date.slice(16, 21)
+        );
+    },
+    //cancels appointment by id if it is valid, forcing a component refresh once list is updated
+    cancelAppointment(id) {
+      AXIOS.delete(CANCEL_APPOINTMENT_ENDPOINT + id, {
+        headers: {
+          token: this.$root.$data.token
+        }
+      })
+        .then(response => {
+          this.items = [];
+          this.getAppointments();
+          console.log(response);
+        })
+        .catch(error => {
+          if (error.response) {
+            if (error.response.status === 400) {
+              alert(
+                "You can only cancel appointments at least 7 days in  advance!"
```

Comment:
You can probably just ignore this tiny detail, but there's an extra space
```suggestion
                "You can only cancel appointments at least 7 days in advance!"
```